src/main/scala/com/typesafe/sbt/bundle/SbtBundle.scala

```diff
@@ -163,15 +189,33 @@ object SbtBundle extends AutoPlugin {
     val configFile = writeConfig(configTarget, bundleConf.value)
     val bundleMappings =
       configFile.pair(relativeTo(configTarget)) ++ (mappings in bundleTypeConfig).value.map(relParent)
-    val name = (packageName in Universal).value
-    val archive = Archives.makeZip(bundleTarget, name, bundleMappings, Some(name))
-    val archiveName = archive.getName
-    val exti = archiveName.lastIndexOf('.')
-    val hash = Hash.toHex(digestFile(archive))
-    val hashName = archiveName.take(exti) + "-" + hash + archiveName.drop(exti)
-    val hashArchive = archive.getParentFile / hashName
-    IO.move(archive, hashArchive)
-    streams.value.log.info(s"Bundle has been created: $hashArchive")
+    shazar(bundleTarget, (packageName in Universal).value,
+      bundleMappings, streams.value.log, f => s"Bundle has been created: $f")
+  }
+
+  private def createConfiguration(): Def.Initialize[Task[File]] = Def.task {
+    val bundleTarget = (target in BundleConfiguration).value
+    val configurationTarget = (NativePackagerKeys.stage in BundleConfiguration).value
+    def relParent(p: (File, String)): (File, String) =
+      (p._1, configurationName.value + java.io.File.separator + p._2)
+    val configChildren: List[File] = configurationTarget.listFiles().toList
+    val bundleMappings: Seq[(File, String)] = configChildren.flatMap(_.pair(relativeTo(configurationTarget)))
+    shazar(bundleTarget, configurationName.value,
+      bundleMappings, streams.value.log, f => s"Bundle-Configuration has been created: $f")
+  }
+
+  private def shazar(bundleTarget: File,
+    archiveName: String,
+    bundleMappings: Seq[(File, String)],
+    logger: Logger,
+    message: File => String): File = {
+    val archived = Archives.makeZip(bundleTarget, archiveName, bundleMappings, Some(archiveName))
+    val exti = archived.name.lastIndexOf('.')
+    val hash = Hash.toHex(digestFile(archived))
+    val hashName = archived.name.take(exti) + "-" + hash + archived.name.drop(exti)
+    val hashArchive = archived.getParentFile / hashName
+    IO.move(archived, hashArchive)
+    logger.info(message(hashArchive))
```

Comment:
Nice - although you could get away without passing in the logger and have the message function do that. WDYT?